### PR TITLE
fix: always fail dag when shutdown is enabled

### PR DESCRIPTION
**Pull request by isubasinghe**
Ensures that a DAG is marked as failed always when shutdown strategy is enabled.

### CODE REVIEW

File: workflow/controller/dag.go [hunk 1]
```diff
@@ -203,10 +220,11 @@ func (d *dagContext) assessDAGPhase(targetTasks []string, nodes wfv1.Nodes) wfv1
 		}
 	}
 
-	return result
+	return result, nil
 }
 
 func (woc *wfOperationCtx) executeDAG(ctx context.Context, nodeName string, tmplCtx *templateresolution.Context, templateScope string, tmpl *wfv1.Template, orgTmpl wfv1.TemplateReferenceHolder, opts *executeTemplateOpts) (*wfv1.NodeStatus, error) {
+	shutdownStrategy := woc.GetShutdownStrategy().Enabled()
```
Comment (Joibel) [inline workflow/controller/dag.go]: GetShutdownStrategy().Enabled() is a boolean, would prefer to rename `shutdownStrategy` to `isShutdown` as per parameter to assessDagPhase.
Or just grab it when it is used the once

File: workflow/controller/dag.go [hunk 2]
```diff
@@ -140,15 +146,26 @@ func (d *dagContext) assessDAGPhase(targetTasks []string, nodes wfv1.Nodes) wfv1
 		targetTaskPhases[d.taskNodeID(task)] = ""
 	}
 
+	boundaryNode, err := nodes.Get(d.boundaryID)
+	if err != nil {
+		return "", err
+	}
+	visited := map[string]bool{}
```
Comment (Joibel) [inline workflow/controller/dag.go]: Unused map

File: workflow/controller/dag.go [hunk 3]
```diff
@@ -140,15 +146,26 @@ func (d *dagContext) assessDAGPhase(targetTasks []string, nodes wfv1.Nodes) wfv1
...
 	// BFS over the children of the DAG
-	uniqueQueue := newUniquePhaseNodeQueue(generatePhaseNodes(nodes[d.boundaryID].Children, wfv1.NodeSucceeded)...)
+	uniqueQueue := newUniquePhaseNodeQueue(generatePhaseNodes(boundaryNode.Children, wfv1.NodeSucceeded)...)
 	for !uniqueQueue.empty() {
 		curr := uniqueQueue.pop()
+		visited[curr.nodeId] = true
+
+		currNode, err := nodes.Get(curr.nodeId)
```
Comment (Joibel) [inline workflow/controller/dag.go]: Call currNode node here

File: workflow/controller/workflowpod.go [hunk 4]
```diff
@@ -385,9 +385,11 @@ func (woc *wfOperationCtx) createWorkflowPod(ctx context.Context, nodeName strin
 					c.Args = x.Cmd
 				}
 			}
-			c.Command = append([]string{common.VarRunArgoPath + "/argoexec", "emissary",
+			c.Command = append([]string{
```
Comment (Joibel) [inline workflow/controller/workflowpod.go:388]: Formatting change only

**Review by JPZ13 — APPROVED** [review]

**Pull request merged**